File: Backend/src/Services/CatalogService/CatalogService.API/Dtos/ForProduct/ProductDetailDto.cs
using CatalogService.API.Dtos.ForCategory;

namespace CatalogService.API.Dtos.ForProduct
{
    public class ProductDetailDto
    {
        public ProductDetailDto()
        {
            Categories = new HashSet<CategoryDto>();
            Features = new HashSet<ProductFeatureDto>();
            Images = new HashSet<ProductImageDto>();
        }
        public int Id { get; set; }
        public string? Name { get; set; }
        public string? BrandName { get; set; }
        public int BrandId { get; set; }
        public string? Description { get; set; }
        public int UnitsInStock {  get; set; }
        public decimal Price { get; set; }
        public ICollection<CategoryDto>? Categories { get; set; }
        public ICollection<ProductFeatureDto>? Features { get; set; }
        public ICollection<ProductImageDto>? Images { get; set; }
    }
}

File: src/Clients/Web/src/dtos/products/productDetailDto.ts
import { CategoryDto } from "../categories/categoryDto";
import { ProductFeatureDto } from "../features/productFeatureDto";
import { ProductImageDto } from "./productImageDto";

export interface ProductDetailDto {
    id:number;
    name:string;
    description:string;
    brandName:string;
    brandId:number;
    price:number;
    unitsInStock:number;
    categories:CategoryDto[];
    features:ProductFeatureDto[];
    images:ProductImageDto[];
}